locations/container/src/main/java/org/apache/brooklyn/container/location/kubernetes/SafeKubernetesClient.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.container.location.kubernetes;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.dsl.*;
+import org.apache.brooklyn.util.core.config.ResolvingConfigBag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SafeKubernetesClient extends DefaultKubernetesClient {
```

Comment:
I don't think this class should extend DefaultKubernetesClient as you're wrapping an instance of the client - should probably just implement Kubernetes client

Reply:
Also can you add a javadoc explaining what this class is for and add some unit tests please

Reply:
If I implement the interface, there will be a whole lot of methods to provide an implementation for.